src/seismometer/seismogram.py

```diff
@@ -61,29 +57,19 @@ def __init__(
 
         Parameters
         ----------
-        config_path : str or Path, optional
-            Where to find the primary configuration file, defaults to the notebook's data directory.
-        output_path : str or Path, optional
-            Override location to place resulting data and report files.
-            Defaults to the config.yml info_dir, and then the notebook's output directory.
-        definitions : dict, optional
-            Additional definitions to be used instead of loading based on configuration, by default None.
+        config : ConfigProvider
+            The configuration provider instance.
+        dataloader : SeismogramLoader, optional
+            A loader instance for defining the data loading pipeline.
 
         """
-        if config_path is None:
-            config_path = Path.cwd() / "data"
-        else:
-            config_path = Path(config_path)
+        self.config = config
+        self.dataloader = dataloader
 
         self.dataframe: pd.DataFrame = None
         self.cohort_cols: list[str] = []
-        self.config_path = config_path
-
-        self.load_config(config_path, definitions=definitions)
 
-        self.config.set_output(output_path)
-        self.config.output_dir.mkdir(parents=True, exist_ok=True)
-        self.dataloader = loader_factory(self.config)
+        self.copy_config_metadata()
```

Comment:
should we copy the config metadata, or just have properties fall through to metadata?